### PR TITLE
Integrates Electron + Improvements/Fixes

### PR DESCRIPTION
**Pull request by sean-perkins**
**Bug Fixes**
- Resolves `symlink` script not working on Windows

**Features**
- Implements Electron
- Extends `symlink` script to handle extensions 

### Electron Process
- Symlink the main app folders into an `electron` folder
- Extend build process to target `.desktop`, similar to `.tns` for {N}
- Task hooks to replace: `moduleId: module.id` references that conflict with Electron
- Uses AngularCLI's multiple app syntax, to build a webpack build using all standard tooling
- Injects electron run script into the `dist` directory after it's built
- Launches the `electron` task via npm (doesn't require global install of electron)

Note: This is just the first stab at getting electron integrated. Still need to integrate custom build tasks for: linux, windows, mac, etc.

### CODE REVIEW

File: electron/src/assets [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/seanperkins/Documents/Projects/angular-native-seed/src/assets
```
Comment (db3dev) [inline electron/src/assets]: Symlinks need to be in the gitignore.

Comment (sean-perkins) [inline electron/src/assets]: Yeah and we should probably leave this PR open for a while longer. At least until I solve the ability to actually generate the builds in Electron. I'm working through these conflicts in an Angular Electron seed project. Would be great to get multi-window support out of the gate.

**Comment by db3dev** [comment]
We should take the opportunity with this PR to add post install scripts from root that installs the nativescript and electron project libraries as well.

**Comment by enjoiful** [comment]
+1 FWIW... electron support would be amazing.

**Pull request closed without merging**